api/src/wfl/module/covid.clj

```diff
@@ -229,21 +202,41 @@
         (set/rename-keys (set/map-invert tdr-source-serialized-fields)))
     (throw (ex-info "source_items is not an integer" details))))
 
-(defn ^:private create-tdr-source [tx id request]
-  (let [create  "CREATE TABLE %s OF TerraDataRepoSourceDetails (PRIMARY KEY (id))"
-        details (format "TerraDataRepoSourceDetails_%09d" id)
-        _       (jdbc/execute! tx [(format create details)])
-        items   (-> (select-keys request (keys tdr-source-serialized-fields))
-                    (set/rename-keys tdr-source-serialized-fields)
-                    (assoc :details details)
-                    (->> (jdbc/insert! tx tdr-source-table)))]
-    [tdr-source-type (-> items first :id str)]))
+(defn ^:private peek-tdr-source-details
+  "Get first unconsumed snapshot record from DETAILS table."
+  [{:keys [details] :as _source}]
+  (let [query "SELECT *
+               FROM %s
+               WHERE consumed IS NULL
+               AND snapshot_id IS NOT NULL
+               ORDER BY id ASC LIMIT 1"]
+    (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
+      (->> (format query details)
+           (jdbc/query tx)
+           first))))
+
+(defn ^:private peek-tdr-source-queue
+  "Get first unconsumed snapshot from SOURCE queue."
+  [source]
+  (if-let [{:keys [snapshot_id] :as _record} (peek-tdr-source-details source)]
+    (datarepo/snapshot snapshot_id)))
+
+(defn ^:private pop-tdr-source-queue
+  "Consume first unconsumed snapshot record in DETAILS table, or throw if none."
+  [{:keys [details] :as source}]
+  (if-let [{:keys [id] :as _record} (peek-tdr-source-details source)]
+    (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
+      (let [now (OffsetDateTime/now)]
```

Comment:
It looks like we use vanilla `(OffsetDateTime/now)` across the DB updates? If so, I will update my code wherever used:
```
(.format (OffsetDateTime/now (ZoneId/of "UTC")) (DateTimeFormatter/ofPattern "yyyy-MM-dd HH:mm:ss"))
```
to `(OffsetDateTime/now)` too.

Reply:
That's how I've seen it used throughout, but if we want to enforce a different way we can break it out somewhere common.